measureme/src/profiler.rs

```diff
@@ -111,4 +113,46 @@ impl<S: SerializationSink> Profiler<S> {
                 bytes.copy_from_slice(raw_event_bytes);
             });
     }
+
+    /// Creates a "start" event and returns a `TimingGuard` that will create
+    /// the corresponding "end" event when it is dropped.
+    pub fn start_event<'a>(
+        &'a self,
+        event_kind: StringId,
+        event_id: StringId,
+        thread_id: u64,
+    ) -> TimingGuard<'a, S> {
+        self.record_event(event_kind, event_id, thread_id, TimestampKind::Start);
+
+        TimingGuard(TimingGuardInternal {
+            profiler: self,
+            event_id,
+            event_kind,
+            thread_id,
+        })
+    }
+}
+
+/// When dropped, this `TimingGuard` will record an "end" event in the
+/// `Profiler` it was created by.
+#[must_use]
```

Comment:
Why must_use? That would mean that you would have to manually call `std::mem::drop` to prevent a compiler warning.

Reply:
`let _guard = ...` also works without a warning. This code is basically copied from what we already have in `rustc`:
https://github.com/rust-lang/rust/blob/8e0007f829661e57d008d2e908c95f6e84b04b25/src/librustc/util/profiling.rs#L348-L349

It works well there.

Reply:
And in the compiler version the `#[must_use]` attribute has prevented me from introducing at least two bugs, so I think it's useful `:)`

Reply:
Didnt know that.